src/editors/containers/VideoEditor/components/VideoSettingsModal/components/VideoPreviewWidget/hooks.js

```diff
@@ -0,0 +1,18 @@
+import messages from '../messages';
+
+// https://stackoverflow.com/a/28735961/479084
+const youtubeRegex = /^(?:https?:\/\/)?(?:m\.|www\.)?(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((\w|-){11})(?:\S+)?$/;
+function isYoutubeUrl(url) {
+  return url.match(youtubeRegex) !== null;
+}
+
+function getVideoType(videoSource) {
+  if (isYoutubeUrl(videoSource)) {
+    return messages.videoTypeYoutube;
+  }
+  return messages.videoTypeOther;
+}
```

Comment:
I think this is duplicating [this code](https://github.com/openedx/frontend-lib-content-components/blob/2dc42e6a46f05443032bb98d49c8c6cc33ff3832/src/editors/data/services/cms/api.js#L255). I would add `isYoutubeUrl` function to the `api.js` file, and implement it something like (I actually need the same function in the different ticket ;) ):
```js
export const isYoutubeUrl = (src) => parseYoutubeId(src) !== null;
```

Reply:
Thanks! I didn't know that it existed.